src/Symfony/Component/Routing/DependencyInjection/ServiceRouterLoaderPass.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Routing\DependencyInjection;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\Compiler\ServiceLocatorTagPass;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Reference;
+use Symfony\Component\Routing\Loader\DependencyInjection\ServiceRouterLoaderContainer;
+
+class ServiceRouterLoaderPass implements CompilerPassInterface
```

Comment:
the class should be marked internal + have the same note as `ServiceRouterLoaderContainer`

Reply:
Ok for the internal but not for the note : this Compiler pass will not be removed in 5.0. It will just just be changed to directly inject the private service locator instead of our temporary container.

Reply:
unless we can use the new `tagged_locator` argument type and get rid of this class :)

Reply:
can't we right now btw?

Reply:
I did not know about this new feature. I will check it later this week and let you know.

Reply:
Sucessfully used a tagged locator, to be reviewed!